platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/MerkleDb.java

```diff
@@ -121,6 +123,19 @@ public final class MerkleDb {
     @SuppressWarnings("rawtypes")
     private final AtomicReferenceArray<MerkleDbDataSource> dataSources = new AtomicReferenceArray<>(MAX_TABLES);
 
+    /**
+     * For every table name (data source label) there may be multiple tables in a single
+     * database. One of them is "primary", it's used by some virtual map in the merkle tree. All
+     * others are "secondary", for example, they are used during reconnects (both on the teacher
+     * and the learner sides). Secondary tables are deleted, both metadata and data, when the
+     * corresponding data source is closed. When a primary data source is closed, its data is
+     * preserved on disk.
+     *
+     * Secondary tables are not included to snapshots and aren't written to DB metadata.
+     */
+    @SuppressWarnings("rawtypes")
+    private final Set<Integer> primaryTables = ConcurrentHashMap.newKeySet();
```

Comment:
Just curious, why did we chose this method rather than `new ConcurrentHashSet<>();`?

Reply:
AFAIK, Java doesn't provide a dedicated class for concurrent hash sets

Reply:
You are correct.  I thought it was there, but that's perhaps the only Map implementation without a set (and with this odd method)...